operator/pkg/test/util.go

```diff
@@ -0,0 +1,60 @@
+package test
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/kyma-project/kyma-operator/operator/api/v1alpha1"
+)
+
+func ModuleTemplateFactory(sample string, module v1alpha1.Module,
+	profile v1alpha1.Profile, data unstructured.Unstructured,
+) (*v1alpha1.ModuleTemplate, error) {
+	var moduleTemplate v1alpha1.ModuleTemplate
+	err := readModuleTemplate(module, &moduleTemplate)
+	if err != nil {
+		return &moduleTemplate, err
+	}
+	moduleTemplate.Name = module.Name + "-" + sample
+	moduleTemplate.Labels[v1alpha1.ModuleName] = module.Name
+	moduleTemplate.Labels[v1alpha1.ControllerName] = module.ControllerName
+	moduleTemplate.Labels[v1alpha1.ProfileLabel] = string(profile)
+	moduleTemplate.Spec.Channel = module.Channel
+	switch module.ControllerName {
+	case "manifest":
+		moduleTemplate.Spec.Target = v1alpha1.TargetRemote
+	default:
+		moduleTemplate.Spec.Target = v1alpha1.TargetControlPlane
+	}
+	if data.GetKind() != "" {
+		moduleTemplate.Spec.Data = data
+	}
+	return &moduleTemplate, nil
+}
+
+func readModuleTemplate(module v1alpha1.Module, moduleTemplate *v1alpha1.ModuleTemplate) error {
+	var template string
+	switch module.ControllerName {
+	case "manifest":
```

Comment:
Wouldnt this be an option without introducing our flag actually? So the remote flag would be set by controller label instead?

Reply:
You mean just use one template for all case?

Reply:
Currently in our code we use a flag in the module template to determine wether to install it as a central module or not. Instead we can just base it on the controller. If the controller is manifest, then we know to wrap it in a manifest CR

Reply:
ok, understood, but I think conceptually it's better introduce that target flag, although by current design, the manifest operator is the de facto which can do the distinguish. But to bind this two aspect together maybe a bit rash, it's like a temporary workaround to me, for example, if later we introduce another operator to manage remote modules, this assumption will not fit anymore.